src/App.js

```diff
@@ -25,17 +25,21 @@ const OkendoStarRating = (props) => {
   const widgetContainer = useRef(null);
 
   const initialiseReviewsWidget = () => {
-    window.okeWidgetApi.initWidget(widgetContainer.current);
+    // if window.okeWidgetApi.initWidget is available
+    if (typeof window.okeWidgetApi?.initWidget === 'function' && widgetContainer.current !== null) {
```

Comment:
(That shouldn't be necessary now that the event listener is properly registered.)

Reply:
But it can be as a safety because in Real-world applications we have a lot of logic and things going in `useEffect ` so it may not break the application or cause any error. 
I have faced this issue on the live site after this condition things work fine

Reply:
I believe your issue was because the event was registered on `window` instead of `document`, as explained above.

The presence of `window.okeWidgetApi.initWidget` is checked just before calling `initialiseReviewsWidget`, so rechecking it inside this function isn't necessary.